pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -295,6 +298,20 @@ func (d *namespacedResourcesDeleter) updateNamespaceStatusFunc(namespace *v1.Nam
 	return d.nsClient.UpdateStatus(&newNamespace)
 }
 
+// updateSetDeletionFailureFunc will set an annotation indicating that the namespace controller has failed to delete all content
+func (d *namespacedResourcesDeleter) updateSetDeletionFailureFunc(err error) func(namespace *v1.Namespace) (*v1.Namespace, error) {
+	errorAnnotationName := "namespace-controller.kube-controller-manager.kubernetes.io/deletion-error"
```

Comment:
I'm not sure this is a long term solution, can't we enrich `NamespaceStatus` with the corresponding information ?
Also I agree, this problem has been hitting us hard.